extensions/wikia/IndexingPipeline/PipelineEventProducer.class.php

```diff
@@ -2,68 +2,201 @@
 
 class PipelineEventProducer {
 	const ARTICLE_MESSAGE_PREFIX = 'article';
+	const PRODUCER_NAME = 'MWEventsProducer';
+	const CONTENT = 'content';
+	const ROUTE_ACTION_KEY = '_action';
+	const ROUTE_NAMESPACE_KEY = '_namespace';
+	const ROUTE_CONTENT_KEY = '_content';
+	const ACTION_CREATE = 'create';
+	const ACTION_UPDATE = 'update';
+	const ACTION_DELETE = 'delete';
 	/** @var PipelineConnectionBase */
 	protected static $pipe;
 
+	/**
+	 * @desc Send event to pipeline in old format (message with params inside).
+	 * @param $eventName
+	 * @param $pageId
+	 * @param array $params
+	 */
 	public static function send( $eventName, $pageId, $params = [ ] ) {
-		global $wgCityId;
-		$msg = new stdClass();
-		$msg->cityId = $wgCityId;
-		$msg->pageId = $pageId;
+		$msg = self::prepareMessage( $pageId );
 		$msg->args = new stdClass();
 		foreach ( $params as $param => $value ) {
 			$msg->args->{$param} = $value;
 		}
-		try {
-			self::getPipeline()->publish( implode( '.', [ self::ARTICLE_MESSAGE_PREFIX, $eventName ] ), $msg );
-		} catch ( Exception $e ) {
-			\Wikia\Logger\WikiaLogger::instance()->error( __METHOD__, [
-				'exception' => $e,
-				'event_name' => $eventName
-			] );
-		}
+		self::publish( implode( '.', [ self::ARTICLE_MESSAGE_PREFIX, $eventName ] ), $msg );
 	}
 
-	/** @return PipelineConnectionBase */
-	protected static function getPipeline() {
-		if ( !isset( self::$pipe ) ) {
-			self::$pipe = new PipelineConnectionBase();
+	/**
+	 * @desc Send event to pipeline in new format (with flags, see function prepareRoute).
+	 * @param $action
+	 * @param $id
+	 * @param string $ns
+	 * @param array $data
+	 * @param array $flags
+	 */
+	public static function nSend( $action, $id, $ns = self::CONTENT, $data = [ ], $flags = [ ] ) {
+		$route = self::prepareRoute( $action, $ns, $flags, $data);
+		$msg = self::prepareMessage( $id );
+		foreach ( $data as $key => $value ) {
```

Comment:
I still think that this way of processing is identical to the one used for args in https://github.com/Wikia/app/pull/8862/files#diff-51080aaef54b01dd7a86c59c827a2761R25 
and should be extracted